sdk/monitor/Azure.Monitor.Query/src/MetricsBatchQueryClient.cs

```diff
@@ -0,0 +1,214 @@
+
+        private readonly MetricsBatchRestClient _metricBatchClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/>. Uses the default 'https://metrics.monitor.azure.com/.default' endpoint.
+        /// <code snippet="Snippet:CreateMetricsClient" language="csharp">
+        /// var client = new MetricsQueryClient(new DefaultAzureCredential());
+        /// </code>
+        /// </summary>
+        /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
+        public MetricsBatchQueryClient(TokenCredential credential) : this(credential, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/>. Uses the default 'https://metrics.monitor.azure.com/.default' endpoint.
+        /// </summary>
+        /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
+        /// <param name="options">The <see cref="MetricsBatchQueryClientOptions"/> instance to as client configuration.</param>
+        public MetricsBatchQueryClient(TokenCredential credential, MetricsBatchQueryClientOptions options) : this(_defaultEndpoint, credential, options)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/>.
+        /// </summary>
+        /// <param name="endpoint">The data plane service endpoint to use. For example <c>https://metrics.monitor.azure.com/.default</c> for public cloud.</param>
+        /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
+        /// <param name="options">The <see cref="MetricsQueryClientOptions"/> instance to as client configuration.</param>
+        public MetricsBatchQueryClient(Uri endpoint, TokenCredential credential, MetricsBatchQueryClientOptions options = null)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            options ??= new MetricsBatchQueryClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            var scope = "https://metrics.monitor.azure.com/.default";
+            Endpoint = endpoint;
+
+            var pipeline = HttpPipelineBuilder.Build(options,
+                new BearerTokenAuthenticationPolicy(credential, scope));
+
+            _metricBatchClient = new MetricsBatchRestClient(_clientDiagnostics, pipeline, endpoint);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="MetricsBatchQueryClient"/> for mocking.
+        /// </summary>
+        protected MetricsBatchQueryClient()
+        {
+        }
+
+        /// <summary>
+        /// Gets the endpoint used by the client.
+        /// </summary>
+        public Uri Endpoint { get; }
```

Comment:
This should be virtual for mocking.

Reply:
But also, why are we exposing this?

Reply:
I'm following the existing pattern - it is also exposed in the LogsQueryClient.